src/components/nodes/LabelUpdaterNode.tsx

```diff
@@ -0,0 +1,110 @@
+import { CloseIcon } from "@chakra-ui/icons";
+import { Box, Input } from "@chakra-ui/react";
+import { FormEvent, useEffect, useRef, useState } from "react";
+import { Handle, Position, useReactFlow } from "reactflow";
+import { Row } from "../../utils/chakra";
+import { modifyFluxNodeLabel, modifyFluxNode } from "../../utils/fluxNode";
+import { FluxNodeData } from "../../utils/types";
+
+export function LabelUpdaterNode({
+  id,
+  data,
+  isConnectable,
+}: {
+  id: string;
+  data: FluxNodeData;
+  isConnectable: boolean;
+}) {
+  const { setNodes } = useReactFlow();
+
+  const [renameLabel, setRenameLabel] = useState(data.label);
+  const renameBlockRef = useRef<HTMLInputElement>(null);
+
+  const cancel = () => {
+    setNodes((nodes) =>
+      modifyFluxNode(nodes, {
+        id,
+      })
+    );
+  };
```

Comment:
this seems to do... nothing?

Reply:
It's just setting the node `type` from LabelUpdater to undefined

Reply:
but modifyFluxNode does a spread operator so it doesnt override the existing props no? anyways by nothing i meant u can literally try clicking the cancel button locally and see nothing happens hehe

Reply:
```suggestion
  const cancel = () => {
    setNodes((nodes) =>
      modifyFluxNodeLabel(nodes, {
        id,
        label: data.label
      })
    );
  };

```